src/Raven.Client/Documents/Session/Operations/LoadOperation.cs

```diff
@@ -52,7 +52,7 @@ public LoadOperation ById(string id)
             if (_ids == null)
                 _ids = new[] { id };
 
-            if (_session.IsLoadedOrDeleted(id))
+            if (_session.IsLoadedOrDeleted(id) && (_includes == null))
```

Comment:
This does only part of the work. In particular, id doesn't check that the included documents are already there. 
If they are, we don't need to hit the server at all

Reply:
This is to error prone. Sooner or later someone will do `ById` first and then `Include`. Need to move that logic to `CreateRequest` method.

Reply:
@ayende this actually already works, not treated here, see gist: https://gist.github.com/maximburyak/af6438d1847fa999960b99d2dcd1576d
@ppekrol I think that this a topic of a seperate issue.

Reply:
where is the code that actually handles this?

Reply:
It's in the `CreateRequest` method. I will sit with @efratshenhar on Sunday and we will figure it out.